src/sdl2_compat.c

```diff
@@ -684,18 +720,100 @@ SDL_GetHintBoolean(const char *name, SDL2_bool default_value)
     return SDL3_GetHintBoolean(SDL2_to_SDL3_hint(name), default_value) ? SDL2_TRUE : SDL2_FALSE;
 }
 
-/* FIXME: callbacks may need tweaking ... */
+static void SDLCALL
+hint_callback_wrapper(void *userdata, const char *name, const char *oldValue, const char *newValue)
+{
+    HintCallbackInfo *callback_info = (HintCallbackInfo *) userdata;
+    const char *sdl2OldValue, *sdl2NewValue;
+    bool freeSdl2OldValue, freeSdl2NewValue;
+
+    sdl2OldValue = SDL3_to_SDL2_hint_value(name, oldValue, &freeSdl2OldValue);
+    sdl2NewValue = SDL3_to_SDL2_hint_value(name, newValue, &freeSdl2NewValue);
```

Comment:
`SDL3_to_SDL2_hint_value()` expects the SDL2 hint name, but it looks like it's getting the SDL3 hint name here.

Reply:
Indeed. sdl2 names are available in `((HintCallbackInfo *)userdata)->name`;